PennMobile/General/SwiftUI Views/CustomPopupView.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  CustomPopupView.swift
+//  PennMobile
+//
+//  Created by Christina Qiu on 1/26/24.
+//  Copyright © 2024 PennLabs. All rights reserved.
+//
+
+#if canImport(SwiftUI)
+import SwiftUI
+import PennMobileShared
+#endif
+
+// Global state for the popup
+class PopupManager: ObservableObject {
+    @Published var isShown: Bool = false
+    @Published var image: Image = Image(systemName: "star")
+    @Published var title: String = ""
+    @Published var message: String = ""
+    // Add more properties as needed for buttons and actions
+}
+// Popup View
+struct CustomPopupView: View {
+    @Binding var isShown: Bool
+    var image: Image
+    var title: String
+    var message: String
+    var confirmAction: () -> Void
+    var body: some View {
+        ZStack {
+            // Scrim
+            Color(hex: 0x5F5F64)
```

Comment:
We should avoid hardcoding colors - either use an existing one or create one in the asset catalog.